function-invokers/java-function-invoker/src/test/java/io/sk8s/invoker/java/function/FunctionConfigurationTests.java

```diff
@@ -31,52 +34,61 @@
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.context.junit4.SpringRunner;
 
-import io.sk8s.invoker.java.function.FunctionConfiguration;
-
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertThat;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest(classes = FunctionConfiguration.class)
+@SpringBootTest(classes = { FunctionConfiguration.class, FunctionProperties.class })
 public abstract class FunctionConfigurationTests {
 
-	@TestPropertySource(properties = { "function.jarLocation=file:target/test-classes",
-			"function.className=" +
+	@TestPropertySource(properties = { "function.uri=file:target/test-classes?io.sk8s.invoker.java.function.FunctionConfigurationTests.Doubler"
+	})
+	public static class SingleFunctionTests extends FunctionConfigurationTests {
+	
+		@Autowired
+		@Qualifier("function0")
+		private Function<Integer, Integer> function;
+	
+		@Test
+		public void testDouble() {
+			assertThat(function.apply(2), is(4));
+		}
+	}
+	
+	@TestPropertySource(properties = { "function.uri=file:target/test-classes?" +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.NumberEmitter," +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.Frenchizer"
 	})
 	public static class SupplierCompositionTests extends FunctionConfigurationTests {
 
 		@Autowired
-		@Qualifier("function")
-		private Supplier<String> function;
+		@Qualifier("function0")
+		private Supplier<Integer> function;
```

Comment:
It seems the composition order is backwards now, or am I missing something?

Reply:
I don't know. Maybe I didn't understand the old code. But "function0" is the first function on the list you passed in (hence `NumberEmitter`). Doesn't that make sense?

Reply:
This test no longer tests composition, does it?
Seems the property could only contain `NumberEmitter` (ie drop `Frenchizer`) and it wouldn't change a thing.

Hence, it needs to be renamed IMHO.